### PR TITLE
Fix #919 result.ruleKey -> result.ruleId

### LINKED ISSUE #919: v1 -> v2 transformer doesn't transform result.ruleKey properly
Should prefer ruleKey over ruleId for v2 result.ruleId
result.ruleId = v1result.ruleKey ?? v1result.ruleId

### PR DESCRIPTION
**Pull request by EasyRhin0**

### CODE REVIEW

File: src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs [hunk 1]
```diff
@@ -400,7 +413,21 @@ public void SarifTransformerTests_ToVersionOne_OneRunWithRules()
...
+        {
+          ""ruleId"": ""C2002""
+        },
+        {
+          ""ruleId"": ""C2003""
+        },
+        {
+          ""ruleId"": ""C2003"",
+          ""ruleKey"": ""C2003-1""
+        }
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: Good.

File: src/Sarif.UnitTests/SarifVersionOneToCurrentTests.cs [hunk 2]
```diff
@@ -474,11 +502,14 @@ public void SarifTransformerTests_ToCurrent_OneRunWitRules()
             ""configuration"": {
               ""defaultLevel"": ""note""
             }
+          },
+          ""C2002-1"": {
+            ""id"": ""C2002""
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifVersionOneToCurrentTests.cs]: Good

File: src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs [hunk 3]
```diff
@@ -408,6 +407,24 @@ internal ResultVersionOne CreateResult(Result v2Result)
                     }
                 }
 
+                if (_currentV2Run.Resources?.Rules != null)
+                {
```
Comment (lgolding) [inline src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs]: Bug: If there's no rules dictionary, you'll neglect to populate v1's `result.ruleId`.

Comment (lgolding) [reply src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs]: In that case, just copy over v2's `result.ruleId`.

File: src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs [hunk 4]
```diff
@@ -829,6 +870,19 @@ internal Run CreateRun(RunVersionOne v1Run)
 
                     _currentRun = run;
 
+                    if (v1Run.Rules != null)
```
Comment (lgolding) [inline src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs]: Right -- populate the rules first.

File: src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs (hunk 3 above)
Comment (EasyRhin0) [reply src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs]: Fixed

**Pull request merged**